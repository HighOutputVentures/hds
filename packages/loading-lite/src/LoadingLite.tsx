import { Box, Text } from '@chakra-ui/react';
import React, { FC } from 'react';

import './custom.css';
export interface LoadingLiteProps {
  size?: 'xs' | 'sm' | 'md' | 'lg' | 'xl';
  showLabel?: boolean;
  labelText?: string;
}

const sizes = {
  xs: '20px',
  sm: '30px',
  md: '40px',
  lg: '50px',
  xl: '60px',
};

const LoadingLite: FC<LoadingLiteProps> = ({
  size = 'md',
  showLabel = true,
  labelText = 'Loading...',
}) => {
  return (
    <Box textAlign="center" pointerEvents="none">
      <Box
        margin="auto"
        border="2px solid #f3f3f3"
        borderTop="4px solid #3498db"
        borderRadius="50%"
        width={sizes[size]}
        height={sizes[size]}
        animation="spin 2s linear infinite"
      />
      {showLabel && <Text fontSize="10px">{labelText}</Text>}
    </Box>
  );
};

export default LoadingLite;
