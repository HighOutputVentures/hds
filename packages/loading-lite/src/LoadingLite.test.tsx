import { render } from '@testing-library/react';
import LoadingLite from './LoadingLite';
import * as React from 'react';

describe('LoadingLite', () => {
  it('should render', () => {
    const { container } = render(<LoadingLite />);

    expect(container).toMatchSnapshot();
  });

  it('should show the label by default', () => {
    const { getByText } = render(<LoadingLite />);

    expect(getByText('Loading...')).toBeTruthy();
  });

  it('should update the label', () => {
    const labelText = 'Please wait...';

    const { getByText } = render(<LoadingLite labelText={labelText} />);

    expect(getByText(labelText)).toBeTruthy();
  });

  it('should hide the label', () => {
    const { queryByText } = render(<LoadingLite showLabel={false} />);

    expect(queryByText('Loading...')).toBeFalsy();
  });
});
