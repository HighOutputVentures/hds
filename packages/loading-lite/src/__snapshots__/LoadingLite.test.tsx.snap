// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoadingLite should render 1`] = `
<div>
  <div
    class="css-w3pvmd"
  >
    <div
      class="css-nwcjbf"
    />
    <p
      class="chakra-text css-9c7r58"
    >
      Loading...
    </p>
  </div>
</div>
`;
